x-pack/plugins/endpoint/server/routes/resolver/utils/pagination.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+import { ResolverEvent } from '../../../../common/types';
+import { extractEventID } from './normalize';
+import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
+
+export interface PaginationParams {
+  size: number;
+  timestamp?: Date;
+  eventID?: string;
+}
+
+interface PaginationCursor {
+  timestamp: Date;
+  eventID: string;
+}
+
+function urlEncodeCursor(data: PaginationCursor) {
+  const value = JSON.stringify(data);
+  return Buffer.from(value, 'utf8')
```

Comment:
❔  Would it be useful/safer to use: https://nodejs.org/dist/latest-v12.x/docs/api/string_decoder.html ?

Reply:
I think that we want to handle decoding with exactly what we were handed--if we're handed invalid multibyte characters I think we should hand those off to elasticsearch exactly as is. The `StringDecoder` api seems to try and [infer the intent of invalid multibyte encodings](http://me.dt.in.th/page/StringDecoder) through buffering and I would imagine would actually cause more problems than it would fix.